### ISSUE TITLE
Layout test fast/box-shadow/inset-box-shadows.html cashes

### ISSUE THREAD

**Issue by annulen** [annulen/webkit#28]
The same bug affects rendering of Twitter and JIRA.

**Comment by annulen** [comment]
```
Program received signal SIGSEGV, Segmentation fault.
0x0000000000000020 in ?? ()
(gdb) bt
#0  0x0000000000000020 in ?? ()
#1  0x00007ffff7996558 in QPainter::save() () from /opt/Qt5.4.0/5.4/gcc_64/lib/libQt5Gui.so.5
#2  0x00007ffff4de7278 in WebCore::GraphicsContext::savePlatformState (this=0x7fffdeda72e8)
    at ../../../Source/WebCore/platform/graphics/qt/GraphicsContextQt.cpp:389
#3  0x00007ffff46569d7 in WebCore::GraphicsContext::save (this=0x7fffdeda72e8)
    at ../../../Source/WebCore/platform/graphics/GraphicsContext.cpp:368
#4  0x00007ffff407d364 in WebCore::GraphicsContextStateSaver::GraphicsContextStateSaver (
    this=0x7fffffff9340, context=..., saveAndRestore=true)
    at ../../../Source/WebCore/platform/graphics/GraphicsContext.h:636
#5  0x00007ffff469585c in WebCore::ShadowBlur::drawInsetShadowWithTiling (this=0x7fffffff9550, 
    graphicsContext=..., rect=..., holeRect=..., templateSize=..., edgeSize=...)
    at ../../../Source/WebCore/platform/graphics/ShadowBlur.cpp:683
#6  0x00007ffff4695236 in WebCore::ShadowBlur::drawInsetShadow (this=0x7fffffff9550, 
    graphicsContext=..., rect=..., holeRect=...)
    at ../../../Source/WebCore/platform/graphics/ShadowBlur.cpp:557
#7  0x00007ffff4deb7cf in WebCore::GraphicsContext::fillRectWithRoundedHole (this=0x7fffffffb348, 
    rect=..., roundedHoleRect=..., color=...)
    at ../../../Source/WebCore/platform/graphics/qt/GraphicsContextQt.cpp:1013
#8  0x00007ffff48671ff in WebCore::RenderBoxModelObject::paintBoxShadow (this=0x7fffded898a0, 
    info=..., paintRect=..., style=..., shadowStyle=WebCore::Inset, includeLogicalLeftEdge=true, 
    includeLogicalRightEdge=true) at ../../../Source/WebCore/rendering/RenderBoxModelObject.cpp:2391
#9  0x00007ffff4832326 in WebCore::RenderBox::paintBoxDecorations (this=0x7fffded898a0, paintInfo=..., 
    paintOffset=...) at ../../../Source/WebCore/rendering/RenderBox.cpp:1363
#10 0x00007ffff47c1c89 in WebCore::RenderBlock::paintObject (this=0x7fffded898a0, paintInfo=..., 
    paintOffset=...) at ../../../Source/WebCore/rendering/RenderBlock.cpp:1555
#11 0x00007ffff47c106f in WebCore::RenderBlock::paint (this=0x7fffded898a0, paintInfo=..., 
    paintOffset=...) at ../../../Source/WebCore/rendering/RenderBlock.cpp:1422
#12 0x00007ffff491af17 in WebCore::RenderLayer::paintBackgroundForFragments (this=0x7fffdedbf240, 
    layerFragments=..., context=..., transparencyLayerContext=..., transparencyPaintDirtyRect=..., 
    haveTransparency=false, localPaintingInfo=..., paintBehavior=0, subtreePaintRootForRenderer=0x0)
    at ../../../Source/WebCore/rendering/RenderLayer.cpp:4670
#13 0x00007ffff4917566 in WebCore::RenderLayer::paintLayerContents (this=0x7fffdedbf240, 
    originalContext=..., paintingInfo=..., paintFlags=96)
    at ../../../Source/WebCore/rendering/RenderLayer.cpp:4331
#14 0x00007ffff4916ce5 in WebCore::RenderLayer::paintLayerContentsAndReflection (this=0x7fffdedbf240, 
    context=..., paintingInfo=..., paintFlags=96)
    at ../../../Source/WebCore/rendering/RenderLayer.cpp:3998
#15 0x00007ffff49159e5 in WebCore::RenderLayer::paintLayer (this=0x7fffdedbf240, context=..., 
    paintingInfo=..., paintFlags=96) at ../../../Source/WebCore/rendering/RenderLayer.cpp:3980
#16 0x00007ffff491b074 in WebCore::RenderLayer::paintList (this=0x7fffdedbfb40, list=0x7fffdedfa5f0, 
    context=..., paintingInfo=..., paintFlags=96)
    at ../../../Source/WebCore/rendering/RenderLayer.cpp:4448
```

**Comment by annulen** [comment]
Fixed in 9916dbe6789cf12838361ed15863be13dc1d1ad4
